StudyHub/StudyHub.BLL/Services/Auth/AuthService.cs

```diff
@@ -50,8 +54,8 @@ public async Task<AuthSuccessDTO> RegisterAsync(RegisterUserDTO dto)
         var invitedUser = _invitedUserRepository.FirstOrDefault(e => e.Email == dto.Email)
             ?? throw new NotFoundException($"User with this email wasn't invited: {dto.Email}");
 
-        if (invitedUser.Token != dto.Token)
-            throw new IncorrectParametersException("Passed token isn't valid.");
+        if (!_encryptService.Verify(HttpUtility.UrlDecode(dto.Token), invitedUser.Token))
+            throw new InvalidSecurityAlgorithmException($"User with this token doesn't exist: {dto.Token}");
```

Comment:
change to another exception
and update exception message to say where user is wrong